Model/Indexer/Data.php

```diff
@@ -118,15 +183,75 @@ public function executeRow($id)
      */
     public function execute($ids)
     {
-        $storesWithNosto = $this->nostoHelperAccount->getStoresWithNosto();
-        foreach ($storesWithNosto as $store) {
-            try {
-                $this->nostoServiceIndex->indexProducts($store, $ids);
-                // Catch only MemoryOutOfBoundsException as this is the most expected ones
-                // And the ones we are interested of
-            } catch (MemoryOutOfBoundsException $e) {
-                $this->nostoLogger->error($e->getMessage());
+        $dimensionProvider = $this->getDimensionsProvider();
```

Comment:
Simply call `$this->executeInParallel($dimensionProvider, $ids);` Doesn't matter if `dimensionProvider` is null.

Reply:
Remove these null checks from here.